### PR TITLE
Updated flow for ETA journey

### PR DESCRIPTION
**Pull request by ethax-ross**
### Trello card

[Trello-4126](https://trello.com/c/0oMN9ytN/4126-remove-the-predicted-class-question-from-the-adviser-funnel-for-domestic-etas)
[Trello-4131](https://trello.com/c/DW00IMjc/4131-remove-the-gcse-science-question-from-the-adviser-funnel-for-domestic-etas)
[Trello-4128](https://trello.com/c/n2LpT5D7/4128-remove-the-gcse-english-maths-question-from-the-adviser-funnel-for-domestic-etas)
[Trello-4129](https://trello.com/c/pBwr1e8Y/4129-remove-subjects-interested-in-teaching-question-for-domestic-etas-from-the-adviser-funnel)
[Trello-4127](https://trello.com/c/ciYNVk21/4127-remove-the-stage-interested-in-teaching-questions-from-the-adviser-funnel-for-domestic-etas)
[Trello-4130](https://trello.com/c/fWS7Ugtq/4130-remove-when-to-start-the-teacher-training-question-from-the-adviser-funnel-for-domestic-etas)

### Changes proposed in this pull request

- Skip WhatDegreeClass when studying for a degree

We currently show the `WhatDegreeClass` step for non-returning candidates that either have a degree or are studying for one. Going forward we only want to show this step for candidates that have a degree.

- Skip StageInterestedTeaching when studying for a degree

We currently show the `StageInterestedTeaching` step for all candidates. Going forward we only want to show this step for candidates that have a degree or are returning to teaching (which implies they have a degree).

-  Skip StageOfDegree when studying for a degree

We currently show the `StageOfDegree` step for all candidates. Going forward we only want to show this step for candidates that have a degree or are returning to teaching (which implies they have a degree).

- Skip GCSE steps when studying for a degree

We currently require people who have a degree or are studying for a degree to specify their GCSE
qualifications. Going forward we only want to show this step for candidates that have a degree.

- Skip StartTeacherTraining step for students

We currently require students to specify the year they wish to start teacher training; going forward we are no longer going to ask this.

- Show SubjectInterestedTeaching to non-final year students

We no longer show the StageInterestedTeaching step to non-final year students, which results in the SubjectInterestTeaching step in being skipped (it requires secondary stage interested in teaching). Update to force the step to always show for non-final year students.

### Guidance to review

This can be tested but the submission will fail until this API PR is merged: https://github.com/DFE-Digital/get-into-teaching-api/pull/1144

### CODE REVIEW

**Comment by gemmadallmandfe** [comment]
@ethax-ross Tejas and I have retested - we can see the record going into CRM (so no API issues)

The problem is more at the CRM end in terms of not being displayed on an adviser dashboard. Tejas will investigate and we'll work out whether we have to add any default information

**Comment by gemmadallmandfe** [comment]
@ethax-ross 

This ETA shortened flow applies to students in their **First year, Second year or Other **(year)**.

To ensure that the ETA candidates going through this shortened flow appear on the ETA allocations dashboard in CRM, we need to infer their teacher training start year.

We don't want to re-add this teacher training start date screen for ETA candidates as we are trying to shorten the journey as much as possible for people at grad fairs etc. In addition, this might seem like a high commitment question for ETA candidates who might be only just starting to explore teaching - we don't want to put them off.

Teacher training start year should ideally be inferred for ETA candidates as follows:

- I'm studying for a degree AND **First year** AND Date between 1 September and 31 December = Current calendar year + 3 years
- I'm studying for a degree AND **First year** AND Date between 1 January and 31 August = Current calendar year + 2 years
- I'm studying for a degree AND **Second year** OR **Other** AND Date between 1 September and 31 December = Current calendar year + 2 years
- I'm studying for a degree AND **Second year** OR **Other** AND Date between 1 January and 31 August = Current calendar year + 1 year

NB advisers will be asked to confirm this start year when they have initial conversations with ETA candidates so we don't need to be 100% accurate; we just need to provide enough information so that they appear on the most likely correct allocations dashboard in CRM.

Other candidates (Final year students, those who already have a degree, non-UK citizens) will be asked for their teacher training start year directly as currently so no change needed for them.

(Ultimately, we will probably look to change this which year are you studying question to be which year do you graduate, which will make allocations much easier!)

Happy to discuss before you look at this - and there might be better ways of doing it!

**Comment by gemmadallmandfe** [comment]
@ethax-ross Tejas and I have tested this from a front-end perspective and also to ensure that the first/second/other year records now have inferred start year and appear on the ETA allocations dashboard in CRM - all looks good

The associated CRM change needs to be deployed to production before this PR can be merged. The CRM deployment is currently scheduled for the evening on Monday 20 Feb, so all being well we should be able to merge this first thing on Tues morning.

Presuming this need a tech review in the meantime?

File: app/models/teacher_training_adviser/steps/start_teacher_training.rb [hunk 1]
```diff
@@ -5,6 +5,11 @@ class StartTeacherTraining < GITWizard::Step
     validates :initial_teacher_training_year_id, inclusion: { in: :year_ids }
 
     NOT_SURE_ID = 12_917
+    STUDYING_NOT_FINAL_YEAR = {
+      second_year: 222_750_002,
```
Comment (asatwal) [inline app/models/teacher_training_adviser/steps/start_teacher_training.rb:9]: Might be a good idea to have these constants in a Module that is included higher up in the hierarchy, or just included in the components that need them.

**Review by asatwal — APPROVED** [review]
👍 Looks Good - Just left some suggestions for your perusal.

**Comment by gemmadallmandfe** [comment]
@ethax-ross Tejas and I have retested for First, Second, Other and Final year students - all looks good

CRM team will deploy their changes today and we can then deploy this PR first thing tomorrow morning (once I've checked that the CRM deployment has gone through successfully)

**Review by gemmadallmandfe — APPROVED** [review]
CRM team have confirmed that the deployment on their side was successful last night so happy for this PR to be merged now

**Pull request merged**